File: orange_hrm_ts/tests/pages/LoginPage.ts
import { Page } from '@playwright/test';

export class LoginPage {
  constructor(private page: Page) {}

  async goto() {
    await this.page.goto('/web/index.php/auth/login');
  }

  async login(user: string, pass: string) {
    await this.page.getByPlaceholder('Username').fill(user);
    await this.page.getByPlaceholder('Password').fill(pass);
    await this.page.getByRole('button', { name: 'Login' }).click();
  }

  getError() {
    return this.page.locator('.oxd-alert-content-text');
  }
}

File: orange_hrm_python/tests/pages/login_page.py
class LoginPage:
    def __init__(self, page):
        self.page = page

    def goto(self):
        self.page.goto("https://opensource-demo.orangehrmlive.com")

    def login(self, user, password):
        self.page.get_by_placeholder("Username").fill(user)
        self.page.get_by_placeholder("Password").fill(password)
        self.page.get_by_role("button", name="Login").click()

    def get_error(self):
        return self.page.locator(".oxd-alert-content-text")